php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -2835,8 +2889,38 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         if ($request->getMethod() === "POST" && !isset($req['ClearInstrument'])) {
             $this->save();
         }
-        return (new \Laminas\Diactoros\Response())
+
+        // Disable form if data entry is complete
+        $user      = $request->getAttribute("user");
+        $sessionID = $request->getQueryParams()["sessionID"];
+        $timepoint = \Timepoint::singleton($sessionID);
+
+        // create an instrument status object
+        $status = new \NDB_BVL_InstrumentStatus;
+        $status->select($this->commentID);
+
+        // freeze the form to prevent data entry
+        // IF (1) Instrument is NOT in progress
+        // OR (2) The user does NOT have `data_entry` permission
+        // OR (3) The user does NOT have access to the timepoint
+        //        at which the instrument is instantiated
+        // OR (4) The instrument is NOT in preview mode
+        // OR (5) the instrument is NOT a direct entry
+        if ($status->getDataEntryStatus() != 'In Progress'
+            || !$user->hasPermission('data_entry')
+            || (isset($timepoint)
+            && !in_array(
+                $timepoint->getData("CenterID"),
+                $user->getData("CenterIDs"),
+                true
+            ))
+            || $this->preview !== true
+            || $this->DataEntryType !== 'DirectEntry'
+        ) {
```

Comment:
```suggestion
        if ($status->getDataEntryStatus() != 'In Progress'
            || !$user->hasPermission('data_entry')
            || (isset($timepoint)
            && !in_array(
                $timepoint->getData("CenterID"),
                $user->getData("CenterIDs"),
                true
            ))
            && ($this->preview !== true
            || $this->DataEntryType !== 'DirectEntry')
        ) {
```

Reply:
"Move the if into the same statement", they said.

"That'll make it simpler," they said.